src/Components/AdminTools/AddNewCustomer.tsx

```diff
@@ -39,24 +36,36 @@ const AddCustomer: React.FC<CustomerAdminDataProps> = ({ customerData, editMode
 
   const handleSave = () => {
     const payload = {
-      id: 1,
-      roleId: role?.code,
       Name: name,
       UserId: id,
       Phone: phone,
       Email: email,
       street: street,
       Apt: apt,
-      stateId: state?.code,
-      countryId: country?.code,
       zipCode: zipCode,
       password: password,
-      createdBy: 'System',
-      updatedBy: 'System',
     }
     console.log('Payload:', payload)
+    //Api not implemented yet from backend
   }
 
+  useEffect(() => {
+    const fetchData = async () => {
+      const { rolesData, countriesData, statesData } = await getMetaData()
+      console.log(rolesData, countriesData, statesData)
+      if (rolesData !== null) {
+        setRolesData(rolesData)
+      }
+      if (countriesData !== null) {
+        setCountriesData(countriesData)
+      }
+      if (statesData !== null) {
+        setStatesData(statesData)
+      }
+    }
+    fetchData()
+  }, [])
```

Comment:
1. don't add entire code in use effect, always create functions and call it in useEffect.
2. Memoize / useCallback on this function.
3. add a dependency to to useEffect (useCallback function)